src/control/lib/daos/pool.go

```diff
@@ -120,50 +129,53 @@ type PoolServiceState uint
 
 const (
 	// PoolServiceStateCreating indicates that the pool service is being created
-	PoolServiceStateCreating PoolServiceState = iota
+	PoolServiceStateCreating = PoolServiceState(mgmtpb.PoolServiceState_Creating)
 	// PoolServiceStateReady indicates that the pool service is ready to be used
-	PoolServiceStateReady
+	PoolServiceStateReady = PoolServiceState(mgmtpb.PoolServiceState_Ready)
 	// PoolServiceStateDestroying indicates that the pool service is being destroyed
-	PoolServiceStateDestroying
+	PoolServiceStateDestroying = PoolServiceState(mgmtpb.PoolServiceState_Destroying)
 	// PoolServiceStateDegraded indicates that the pool service is in a degraded state
-	PoolServiceStateDegraded
+	PoolServiceStateDegraded = PoolServiceState(mgmtpb.PoolServiceState_Degraded)
 	// PoolServiceStateUnknown indicates that the pool service state is unknown
-	PoolServiceStateUnknown
+	PoolServiceStateUnknown = PoolServiceState(mgmtpb.PoolServiceState_Unknown)
 )
 
 func (pss PoolServiceState) String() string {
-	if pss < PoolServiceStateCreating || pss > PoolServiceStateUnknown {
-		return fmt.Sprintf("Invalid pool service state %d", pss)
+	psss, ok := mgmtpb.PoolServiceState_name[int32(pss)]
+	if !ok {
+		return "invalid"
 	}
-	return [...]string{
-		mgmtpb.PoolServiceState_Creating.String(),
-		mgmtpb.PoolServiceState_Ready.String(),
-		mgmtpb.PoolServiceState_Destroying.String(),
-		mgmtpb.PoolServiceState_Degraded.String(),
-		mgmtpb.PoolServiceState_Unknown.String(),
-	}[pss]
+	return psss
 }
 
 func (pss PoolServiceState) MarshalJSON() ([]byte, error) {
 	return []byte(`"` + pss.String() + `"`), nil
 }
 
+func unmarshalStrVal(inStr string, name2Vals map[string]int32, val2Names map[int32]string) (int32, error) {
```

Comment:
Given this is essentially the same approach we are using with `NvmeDevState` and `LedState` do we want to formalise it somehow with a helper so we can update any improvements to be approach in one place?

Reply:
Probably not a bad idea to create a helper in the `common/proto` package. I'd like to do that as a follow-on, though.